var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -85,22 +136,40 @@ class Mapl(CMakePackage):
     depends_on("esmf", when="@:2.12.99")
     depends_on("esmf~debug", when="~debug")
     depends_on("esmf+debug", when="+debug")
-    depends_on("yafyaml@:0.5.1")
+
+    depends_on("gftl@1.5.5:")
     depends_on("gftl-shared@1.3.1:")
 
-    # fArgParse not yet available
-    conflicts("+fargparse")
+    # There was an interface change in yaFyaml, so we need to control versions
+    # MAPL 2.22 and older uses older version, MAPL 2.23+ and higher uses newer
+    depends_on("yafyaml@1.0-beta5", when="@:2.22+extdata2g")
+    depends_on("yafyaml@1.0.4:", when="@2.23:+extdata2g")
+
+    # pFlogger depends on yaFyaml in the same way. MAPL 2.22 and below uses old
+    # yaFyaml so we need to use old pFlogger, but MAPL 2.23+ uses new yaFyaml
+    depends_on("pflogger@:1.6", when="@:2.22+pflogger")
+    depends_on("pflogger@1.9.1:", when="@2.23:+pflogger")
+
+    # fArgParse v1.4.1 is the first usable version with MAPL
+    depends_on("fargparse@1.4.1:", when="+fargparse")
+
+    depends_on("pfunit@4.2: +mpi +fhamcrest", when="+pfunit")
+    depends_on("flap", when="+flap")
 
     depends_on("ecbuild", type="build")
 
+    depends_on("python@3:")
+    depends_on("py-numpy", when="+f2py")
+    depends_on("perl")
```

Comment:
Does MAPL always depend on python & perl, regardless of version and variants?

Reply:
@AlexanderRichert-NOAA Well, technically, as a pure library when compiled, no. As used in GEOS, yes.

Python because of the [automatic code generator](https://github.com/GEOS-ESM/MAPL/blob/main/Apps/MAPL_GridCompSpecs_ACG.py) used to generate import/internal/export specs in places like GOCART. The older acg (which might still be used in older parts of GEOS and possibly old parts of GOCART), is Perl.

Now, I suppose technically, this means something like this:
```python
    depends_on("python@3:", type="run")
    depends_on("py-numpy", when="+f2py", type="build")
    depends_on("perl", type="run")
```
since the ACGs are runtime only.

But I don't know if in spack land, this works? I mean, f2py is a build time use of python, so would that mean if you enable it, you'd need python as both build and run. I can't find anywhere in spack where you see a `depends_on` that is "duplicated" as two different types based on a `when`. So at most maybe we could do:

```python
    depends_on("python@3:", type="run", type=("build","run"))
    depends_on("py-numpy", when="+f2py", type="build")
    depends_on("perl", type="run")
```
?

Reply:
Presumably py-numpy depends on python, so I think that would capture the case where +f2py needs python at build time..?

Reply:
Maybe? I'm not sure if spack would bring in python like that or not. I mean, you think so, but then...maybe not?

We could ask the spack devs if it would.

Reply:
For me the main thing was to make sure there weren't any unused dependencies. I'm 99% sure it would work to just use the py-numpy dependency there, but I don't think it's worth getting hung up on either.